mbuild/lib/recipes/polymer.py

```diff
@@ -12,39 +13,61 @@
 class Polymer(Compound):
     """Connect one or more components in a specified sequence.
 
-    Parameters
+    Attributes 
     ----------
-    monomers : mb.Compound or list of mb.Compound
-        The compound(s) to replicate.
-    n : int
-        The number of times to replicate the sequence.
-    sequence : str, optional, default='A'
-        A string of characters where each unique character represents one
-        repetition of a monomer. Characters in `sequence` are assigned to
-        monomers in the order assigned by the built-in `sorted()`.
-    port_labels : 2-tuple of strs, optional, default=('up', 'down')
-        The names of the two ports to use to connect copies of proto.
+    monomers : List of mb.Compounds
+        The compound(s) to replicate. Add to this list using the add_monomers method.
+    end_groups : List of mb.Compounds
+        The compound to cap the end of the polymer. Add to this list using the
+        add_end_groups method. 
+
+    Methods
+    -------
+    add_monomer(monomer, bonding_indices, separation, port_labels, orientation, replace)
+        Use to add a monomer compound to Polymer.monomers
+
+    add_end_groups(compound, bond_index, separation, orientation, replace)
+        Use to add an end group compound to Polymer.end_groups
 
+    build(n, sequence)
+        Use to create a single polymer compound. This method uses the compounds created 
+        by calling the add_monomer and add_end_group methods.
     """
-    def __init__(self, monomers, n, sequence='A', port_labels=('up', 'down')):
+    def __init__(self):
```

Comment:
Thinking about this a little more, what if you had to pass in monomers with two ports (e.g., 'up' and 'down') 
the `add_monomer` could be changed to `create_monomer` and its function could be to replace specific atoms with ports or to add ports. This would allow for the existing CH2() and CH3() classes to be used more easily.

Same for the end_groups, this would allow them to be passed in with the init.

Reply:
I'm not opposed to having `monomers` and `end_groups` as parameters of the `init` function. I think that could be useful for someone that wants to go through the process of creating their own ports instead of using `add_monomer` and `add_end_groups`, and basically going straight to calling the `build` function.  There could be a check that asserts the required ports are in place.

Technically, the `CH2` class does still work with the planned API as is:

```
from mbuild.lib.moieties import ch2
chain = Poly.Polymer()
chain.add_monomer(compound = ch2.CH2(),
                 bonding_indices = [0, 0],
                 separation=0.15,
                 orientation = [(0,1,0), (0,-1,0)],
                 replace=False)
chain.build(n=5)
```

When I run this, it basically reproduces what `alkane.py` does.

Reply:
There may be an issue with this: CH2() already has ports and add_monomer adds more ports on top of them, which I think is not ideal? 
```
chain.visualize(show_ports=True).show()
print(len(chain.all_ports()))  # 10
print(len(chain.available_ports()))  # 0
```

Reply:
Yeah, I think having the option to skip the `add_monomer` and `add_end_groups` functions and go straight to `build` while passing in compounds that already have ports is a good idea.

In regards to the mbuild library moieties, I think it would be nice to have the option to not include ports. Something like `ch2 = CH2(add_ports=False)` but maybe that can be a separate issue, and the changes to the `Polymer` class can be made to accommodate compounds with preexisting ports.